src/NuGetPackageVerifier/Rules/AssemblyIsBuiltInReleaseConfigurationRule.cs

```diff
@@ -1,45 +1,90 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.IO;
 using System.Linq;
-using Mono.Cecil;
-using Mono.Collections.Generic;
-using NuGetPackageVerifier.Logging;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
 
 namespace NuGetPackageVerifier.Rules
 {
-    public class AssemblyIsBuiltInReleaseConfigurationRule : AssemblyHasAttributeRuleBase
+    public class AssemblyIsBuiltInReleaseConfigurationRule : IPackageVerifierRule
     {
-        // TODO remove
-        public override IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
+        // TODO: Revert to using Mono.Cecil when https://github.com/jbevain/cecil/issues/306 is fixed.
+        public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
         {
-            // todo remove "Cecil throws ArgumentException when evaluating constructor arguments"
-            context.Logger.Log(LogLevel.Info, $"{nameof(AssemblyIsBuiltInReleaseConfigurationRule)} skipped");
-            return Enumerable.Empty<PackageVerifierIssue>();
-        }
-
-        public override IEnumerable<PackageVerifierIssue> ValidateAttribute(string currentFilePath, AssemblyDefinition assembly, Collection<CustomAttribute> assemblyAttributes)
-        {
-            if (!HasReleaseConfiguration(assemblyAttributes))
+            foreach (var currentFile in context.PackageReader.GetFiles())
             {
-                yield return PackageIssueFactory.AssemblyHasIncorrectBuildConfiguration(currentFilePath);
+                var extension = Path.GetExtension(currentFile);
+                if (extension.Equals(".dll", StringComparison.OrdinalIgnoreCase) ||
+                    extension.Equals(".exe", StringComparison.OrdinalIgnoreCase))
+                {
+                    var assemblyPath = Path.ChangeExtension(
+                        Path.Combine(Path.GetTempPath(), Path.GetTempFileName()), extension);
+
+                    try
+                    {
+                        using (var packageFileStream = context.PackageReader.GetStream(currentFile))
+                        using (var fileStream = new FileStream(assemblyPath, FileMode.Create))
+                        {
+                            packageFileStream.CopyTo(fileStream);
+                        }
+
+                        if (AssemblyHelpers.IsAssemblyManaged(assemblyPath))
```

Comment:
Can we try catch the `InvalidPEAssemblyException` and continue instead of extracting the file?